File: Data-Structure-I/74-Search-a-2D-Matrix.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        int row = matrix.size(), col = matrix[0].size() - 1, i = 0;
        while(i<row && col>-1){
            if(matrix[i][col] == target) return true;
            else if(matrix[i][col] > target) col--;
            else i++;
        }
        return false;
    }
};

File: Data-Structure-I/74-Search-a-2D-Matrix.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        row, col, i = len(matrix), len(matrix[0]) - 1, 0
        while i<row and col > -1:
            if matrix[i][col] == target:
                return True
            elif matrix[i][col] > target:
                col -= 1
            else:
                i += 1
        return False
